Library/Homebrew/cmd/update.sh

```diff
@@ -685,6 +685,21 @@ EOS
 
   safe_cd "${HOMEBREW_REPOSITORY}"
 
+  # HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+  # who merge homebrew-core into linuxbrew-core to avoid being migrated
+  # We only migrate HOMEBREW_DEVELOPER users right now
+  # Don't migrate on CI either
+  if [[ -n "${HOMEBREW_DEVELOPER}" && -z "${HOMEBREW_CORE_MERGE_MAINTAINER}" && -z "${GITHUB_ACTIONS}" ]]; then
+    HOMEBREW_CORE_REPOSITORY_ORIGIN="$(git -C "${HOMEBREW_LIBRARY}/Taps/homebrew/homebrew-core" remote get-url origin)"
+    if [[ "${HOMEBREW_CORE_REPOSITORY_ORIGIN}" = "https://github.com/Homebrew/linuxbrew-core" ]]; then
+      git -C "${HOMEBREW_LIBRARY}/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core > /dev/null
```

Comment:
Would be nice to share or be done at the same time as the logic here:
https://github.com/Homebrew/brew/blob/a7c8ed09f97e375d18492d2d545a1d4342e72097/Library/Homebrew/cmd/update.sh#L461-L478